cmd/code.go

```diff
@@ -1,133 +1,9 @@
-	ctx := cmd.Context()
-
-	execCh := make(chan types.Exec)
-	errCh := make(chan error)
-	isNew := false
-
-	var err error
-	var execRef string // Can be execID or name
-
-	if config.CreateExec {
-
-		ui.Infof("Initializing node...")
-
-		execCh, errCh, err = execCreateAndWatch(ctx, types.ExecConfig{}, types.GitConfig{})
-		if err != nil {
-			return err
-		}
-		isNew = true
-
-	} else {
-		var createNew bool
-
-		if execRef == "" {
-			execRef, createNew, err = sessionSelectSSHExecRef(cmd, execRef, false)
-			if err != nil {
-				return err
-			}
-		}
-
-		if createNew {
-			execCh, errCh, err = execCreateAndWatch(ctx, types.ExecConfig{}, types.GitConfig{})
-			if err != nil {
-				return err
-			}
-			isNew = true
-		} else {
-			execCh, errCh, err = session.Wait(ctx, execRef)
-			if err != nil {
-				return err
-			}
-		}
-	}
-	for {
-		select {
-		case e := <-execCh:
-			if e.Status == types.StatusRunning {
-
-				if e.Connection == nil {
-					ui.Errorf("❌ Something unexpected happened. No connection info found for session %q", e.ID)
-					ui.Infof("Run `unweave ls` to see the status of your session and try connecting manually.")
-					os.Exit(1)
-				}
-				ui.Infof("🚀 Session %q up and running", e.ID)
-				ui.Infof("🔧 Setting up VS Code ...")
-
-				if err := ssh.RemoveKnownHostsEntry(e.Connection.Host); err != nil {
-					// Log and continue anyway. Most likely the entry is not there.
-					ui.Debugf("Failed to remove known_hosts entry: %v", err)
-				}
-
-				if err := ssh.AddHost("uw:"+e.ID, e.Connection.Host, e.Connection.User, e.Connection.Port); err != nil {
-					ui.Debugf("Failed to add host to ssh config: %v", err)
-				}
-
-				defer func() {
-					if e := ssh.RemoveHost("uw:" + e.ID); e != nil {
-						ui.Debugf("Failed to remove host from ssh config: %v", e)
-					}
-				}()
-
-				// TODO we should wait until port is open
-
-				if !config.NoCopySource && isNew {
-					dir, err := config.GetActiveProjectPath()
-					if err != nil {
-						ui.Errorf("Failed to get active project path. Skipping copying source directory")
-						return fmt.Errorf("failed to get active project path: %v", err)
-					}
-
-					if err := copySource(e.ID, dir, "/home/ubuntu", *e.Connection, ""); err != nil {
-						fmt.Println(err)
-					}
-				} else {
-					ui.Infof("Skipping copying source directory")
-				}
-
-				arg := fmt.Sprintf("vscode-remote://ssh-remote+%s@%s/home/ubuntu", e.Connection.User, e.Connection.Host)
-
-				codeCmd := exec.Command("code", "--folder-uri="+arg)
-				codeCmd.Stdout = os.Stdout
-				codeCmd.Stderr = os.Stderr
-
-				if e := codeCmd.Run(); e != nil {
-					ui.Errorf("Failed to start VS Code: %v", e)
-					os.Exit(1)
-				}
-				ui.Successf("✅ VS Code is ready!")
-				return nil
-			}
-
-		case err := <-errCh:
-			var e *types.Error
-			if errors.As(err, &e) {
-				uie := &ui.Error{Error: e}
-				fmt.Println(uie.Verbose())
-				os.Exit(1)
-			}
-			return err
-
-		case <-ctx.Done():
-			return nil
-		}
-	}
```

Comment:
If the OpenVSCode flag is `true` in `sessionConnect` then this portion of the code will be executed  in its own private method.

Reply:
The command to connect to vscode should probably not be hidden inside sessionConnect. Since this function handles the code specific logic, it might be better to wait on a `ready` chan and then execute that the connection command (`vscode-remote://ssh-remote+%s@%s/home/ubuntu`) here.

Reply:
Resolved